### PR TITLE
Add support for arm64 architecture

### PR DESCRIPTION
**Pull request by ThyMYthOS**

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -22,8 +23,12 @@ apps:
 parts:
   eclipse:
     plugin: dump
-    source: https://download.eclipse.org/technology/epp/downloads/release/2022-09/R/eclipse-java-2022-09-R-linux-gtk-x86_64.tar.gz
-    source-checksum: "sha512/51e8c4d29ce0c0162998d6dd1847a030ce6210e3183fe56f79b535ed841e254ce1fea62bd01eac7d7c09f9c75309abca08d18b30c0de9c931893d57fe834991f"
+    source:
+      - to amd64: "https://download.eclipse.org/technology/epp/downloads/release/2022-09/R/eclipse-java-2022-09-R-linux-gtk-x86_64.tar.gz"
+      - to arm64: "https://download.eclipse.org/technology/epp/downloads/release/2022-09/R/eclipse-java-2022-09-R-linux-gtk-aarch64.tar.gz"
```
Comment (Mailaender) [inline snap/snapcraft.yaml]:
```suggestion
      - on amd64: "https://download.eclipse.org/technology/epp/downloads/release/2022-09/R/eclipse-java-2022-09-R-linux-gtk-x86_64.tar.gz"
      - on arm64: "https://download.eclipse.org/technology/epp/downloads/release/2022-09/R/eclipse-java-2022-09-R-linux-gtk-aarch64.tar.gz"
```
?

**Comment by merlijn-sebrechts** [comment]
This fails to build and has conflicts with the main branch. @ThyMYthOS do you want to continue working on it, or should we close this PR?

**Review by merlijn-sebrechts — DISMISSED** [review]
This fails to build and has conflicts with the main branch. @ThyMYthOS do you want to continue working on it, or should we close this PR?

**Comment by ThyMYthOS** [comment]
The version is outdated anyway. I will give it another try using your suggestions.

**Comment by ThyMYthOS** [comment]
@merlijn-sebrechts seems that `source-checksum` does not support advanced grammar. Any idea how to solve this?

**Comment by merlijn-sebrechts** [comment]
It seems like it's building on X86_64!

Have you tested this on `arm` yourself? If you don't have an arm device, you can use remote-build to build on the launchpad builders from your local machine: https://snapcraft.io/docs/remote-build

Alternatively, you could modify the GitHub action to also build on arm, like we do for gimp: https://github.com/snapcrafters/gimp/blob/master/.github/workflows/build-prs.yml

How do you want to proceed?

**Review by jnsgruk — APPROVED** [review]
I just built this locally for amd64, and also tested `remote-buiild` for `arm64` and it's all good! I'll merge this and follow up another time with a CI uplift to follow the latest actions. Thanks!

Edit: heh, I can't merge unless there is another review.

Build logs for arm64 attached.
[eclipse_arm64.txt](https://github.com/snapcrafters/eclipse/files/13688074/eclipse_arm64.txt)

**Pull request merged**